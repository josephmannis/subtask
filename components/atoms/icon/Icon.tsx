
import React from 'react';
const searchIcon = require('../../../assets/ic_search.png');
const checkIcon = require('../../../assets/ic_check.png')
import styled from 'styled-components/native';

type IconType = 'search' | 'check';
type IconSize = 'small';

interface IIconProps {
    type: IconType;
    size: IconSize;
}

const mapTypeToIcon = (type: IconType) => {
    switch(type) {
        case 'search':
            return searchIcon
        case 'check':
            return checkIcon
    }
}

const mapSize = (size: IconSize) => {
    switch(size) {
        case 'small':
            return '20px'
    }
}

interface ImageProps {
    size: IconSize;
}

const IconImage = styled.Image<ImageProps>`
    height: ${({size}) => mapSize(size)};
    width: ${({size}) => mapSize(size)};
    aspect-ratio: 1;
    resize-mode: contain;
`

const Icon: React.FC<IIconProps> = ({type, size}) => {
    return (
        <IconImage source={mapTypeToIcon(type)} size={size}/>
    )
}

export default Icon;