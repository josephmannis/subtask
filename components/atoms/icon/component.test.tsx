import React from 'react';

import renderer from 'react-test-renderer';
import Icon from './Icon';
test('Renders search', () => {
    const tree = renderer.create(<Icon type='search' size='small'/>).toJSON();
    expect(tree).toMatchSnapshot()
})

test('Renders check', () => {
    const tree = renderer.create(<Icon type='check' size='small' />).toJSON();
    expect(tree).toMatchSnapshot()
})

test('Renders arrow', () => {
    const tree = renderer.create(<Icon type='arrow' size='small' />).toJSON();
    expect(tree).toMatchSnapshot()
})

test('Renders plus', () => {
    const tree = renderer.create(<Icon type='plus' size='small' />).toJSON();
    expect(tree).toMatchSnapshot()
})

test('Renders kebab', () => {
    const tree = renderer.create(<Icon type='kebab' size='small' />).toJSON();
    expect(tree).toMatchSnapshot()
})