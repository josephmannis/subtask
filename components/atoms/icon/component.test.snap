// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders arrow 1`] = `
<Image
  size="small"
  source={1}
  style={
    Array [
      Object {
        "aspectRatio": 1,
        "height": 20,
        "resizeMode": "contain",
        "width": 20,
      },
    ]
  }
/>
`;

exports[`Renders check 1`] = `
<Image
  size="small"
  source={1}
  style={
    Array [
      Object {
        "aspectRatio": 1,
        "height": 20,
        "resizeMode": "contain",
        "width": 20,
      },
    ]
  }
/>
`;

exports[`Renders kebab 1`] = `
<Image
  size="small"
  source={1}
  style={
    Array [
      Object {
        "aspectRatio": 1,
        "height": 20,
        "resizeMode": "contain",
        "width": 20,
      },
    ]
  }
/>
`;

exports[`Renders plus 1`] = `
<Image
  size="small"
  source={1}
  style={
    Array [
      Object {
        "aspectRatio": 1,
        "height": 20,
        "resizeMode": "contain",
        "width": 20,
      },
    ]
  }
/>
`;

exports[`Renders search 1`] = `
<Image
  size="small"
  source={1}
  style={
    Array [
      Object {
        "aspectRatio": 1,
        "height": 20,
        "resizeMode": "contain",
        "width": 20,
      },
    ]
  }
/>
`;
